File: 01-SRP/CPP/FAIL/main.cpp
/**
* El código viola el SRP porque la clase Empleado tiene dos razones para cambiar: 
* si cambian los datos del empleado (nombre/salario) o si cambia la forma en que 
* se persiste (guardar en un archivo, base de datos, etc.).
*/
#include <iostream>
#include <fstream>
#include <string>
#include <memory>

class Empleado 
{
    public:
        std::string nombre;
        double salario;

        Empleado(std::string n, double s) : nombre(n), salario(s) {}

        // Responsabilidad 1: gestionar datos del empleado
        std::string getNombre() const { return nombre; }
        double getSalario() const { return salario; }

        // Responsabilidad 2: guardar en archivo (violación de SRP)
        void guardarEnArchivo() 
        {
            std::ofstream file("empleados.txt", std::ios::app);
            file << nombre << "," << salario << "\n";
            file.close();
            std::cout << "Empleado guardado en archivo.\n";
        }
};

int main()
{
    auto juan = std::make_shared<Empleado>("Ana", 30.5);
    juan->guardarEnArchivo();

    return 0;
}

File: 01-SRP/TYPESCRIPT/FAIL/main.ts
// 1. Importación Correcta: Usamos 'import' para módulos Node.js/CommonJS en TS.
//    También necesitamos instalar los tipos de Node.js (@types/node) para que esto funcione.
import * as fs from 'fs';

class Empleado 
{
    constructor(public nombre: string, public salario: number) {}

    getNombre(): string { return this.nombre; }
    getSalario(): number { return this.salario; }

    guardarEnArchivo(): void 
    {
        const fs = require('fs');
        fs.appendFileSync('empleados.txt', `${this.nombre},${this.salario}\n`);
        console.log('Empleado guardado en archivo.');
    }
}

function main()
{
    let juan = new Empleado("Ana", 30.5);
    juan.guardarEnArchivo();
}

main();